### PR TITLE
Initial commit of Helm charts and e2e testing

### PR DESCRIPTION
**Pull request by kylecrawshaw**
This PR aims to improve validation of changes to opencbdc by setting up a Kubernetes cluster with minikube in Github actions and then install the 2PC architecture. After installation into the minikube cluster there are tests run to check whether or not pods/containers have started and are running. This is a very minimal test that currently does not check anything besides pod/container status. While we are using the Helm chart to run a test this exact configuration can be used to run in any Kubernetes cluster/environment. Documentation should be improved in a future PR to make it clear how to use the Helm chart and customize configuration.

The following has been added:
- Helm Chart for 2PC architecture to simplify running 2PC in Kubernetes
- Go tests using testify and Terratest to install the 2PC Helm Chart and check if containers are running

Signed-off-by: Kyle Crawshaw <kyle.crawshaw@gmail.com>

### CODE REVIEW

File: charts/opencbdc-2pc/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# opencbdc-2pc
```
Comment (kylecrawshaw) [inline charts/opencbdc-2pc/README.md:1]: Auto-generated file using [helm-docs](https://github.com/norwoodj/helm-docs)

**Comment by kylecrawshaw** [comment]
@HalosGhost @kevinkarwaski This PR should now be ready for review.

**Comment by HalosGhost** [comment]
@kevinkarwaski as I have much less familiarity with k8s, I'd really appreciate a pretty thorough double-check from you when you have time!

**Review by kevinkarwaski — APPROVED** [review]
This is a great foundation to build and improve our approach to testing and basic component functional evaluation / assurance.

**Comment by crberube** [comment]
This looks good! Functionally I don't see any issues here, though I have some thoughts and resources on a few areas that I think could be useful to keep in mind.

# Error handling

Generally speaking I have found it best to bubble up errors through to the caller, wrapping each error along the way up to the top of the stack. This helps trace errors and keep error handling logic consistent.

E.g.

```go
package main

import (
    "fmt"
    "log"
)

func main() {
  fmt.Println("Doing some logic...")

  err := doSomeLogging()

  // handle err here
  if err != nil {
    log.Fatalf("we tried. then this happened: %v", err)
  }

}

func doSomeLogging(args ...string) error {
    fmt.Println("Logging some stuff...")
    _, err := badNewsBearsFunc()
    
    if err != nil {
        return fmt.Errorf("error occurred while logging: %s", err)
    }

    fmt.Println("sweet success!")

    return nil
}
```

I have found this to be a great in-depth [article on the subject](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully).

# Testing:

As these tests evolve, it looks like they will be a great candidate for table-driven testing. Here is a [short article](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go) that introduces the concept and here is [another](https://adlerhsieh.com/blog/a-deep-dive-into-table-driven-testing-in-golang) that is a little more in-depth. They can really help with the readability and maintanability when you are testing different inputs against expected outputs on the same function.

# Package visibility

As i'm sure you have seen whether a name starts with a capital or lowercase letter determines visibility in Go. Digital Ocean has a [good article](https://www.digitalocean.com/community/tutorials/understanding-package-visibility-in-go) on it. It appears like we could get away with less exposure here as most of this code seems to be all internal to the test package. Along with that, go linting tools will be pretty strict that anything exported, i.e. "public" has a comment (structs, functions, methods, etc.) so that the `godoc` feature can automatically create docs for it. I've found it be be pretty useful for a minimal amount of extra work.

**Comment by kylecrawshaw** [comment]
@crberube I've update error handling based on your recommendations and I've also switched to using table testing for the test cases.

@HalosGhost I think we're in good shape here.

**Review by crberube — APPROVED** [review]
LGTM

**Comment by HalosGhost** [comment]
@kylecrawshaw I think something may have gone funky; the E2E test takes *much* longer to complete (actually, it seems to hang for quite a while; I got impatient after about 10 minutes and killed it). I assume this is only happening for me at the moment. Let's plan to chat this week to figure out what may be going wrong.

**Comment by kylecrawshaw** [comment]
> @kylecrawshaw I think something may have gone funky; the E2E test takes _much_ longer to complete (actually, it seems to hang for quite a while; I got impatient after about 10 minutes and killed it). I assume this is only happening for me at the moment. Let's plan to chat this week to figure out what may be going wrong.

@HalosGhost I'll take a look at this today and see what is going on.

**Comment by HalosGhost** [comment]
@kylecrawshaw This looks great and works lovely locally. Looks like there's a missing DCO and a linting failure, and then I'll be ready to merge!

**Comment by kylecrawshaw** [comment]
@HalosGhost DCO issue has been resolved and I've fixed the linting error as well. This should be good to now.

**Review by HalosGhost — APPROVED** [review]
Adds a complete, end-to-end test, and the infrastructure work necessary to extend system tests easily (plus a lot of the work necessary to run the code in kubernetes). Looks fabulous to me.

@kylecrawshaw if you would squash to a single commit, I'm ready to merge when you are.

**Pull request merged**